### ISSUE TITLE
Project: Moped v2.0

### ISSUE THREAD

**Issue by johnclary** [cityofaustin/atd-data-tech#10579]
We're expanding the capabilities of our mobility project management solution to support key workgroups.
![moped thumbnail](https://user-images.githubusercontent.com/14793120/197806214-aaffcc19-dcb2-4b47-8ca5-11db42ec6f15.png)
## Background

[Moped](https://www.austinmobility.io/products/5086) is our shared platform for Austin Transportation and partners to manage the City of Austin's complete portfolio of mobility infrastructure projects. We launched [Moped v1.0](https://www.austinmobility.io/projects/307) in early 2022, with a focus on traffic signal projects. Since then, we have continued to [release new features](https://github.com/cityofaustin/atd-moped/releases) on a bi-weekly basis. 

Moped v2.0 encompasses new capabilities to support a much larger portfolio of projects, and crucially, will entail the migration of a legacy project tracking tool into Moped.

## Scope

- A new user interface for project mapping with an improved data structure designed to accommodate projects of all scopes and sizes
- An improved [activity feed](https://github.com/cityofaustin/atd-data-tech/issues/6496) that allows users to easily keep track of project activities
- A data migration of 2,000+ projects from a legacy Microsoft Access database into Moped
- Building a public-facing [mobility project explorer](https://github.com/cityofaustin/atd-data-tech/issues/1106). 

## Screenshots
![Moped-Map](https://github-production-user-asset-6210df.s3.amazonaws.com/1463708/269081007-aca376f8-149c-40ad-b3b6-8769b669b228.png)
![Moped-Projects](https://github-production-user-asset-6210df.s3.amazonaws.com/1463708/269081010-b192a325-a134-4846-a263-8c92471eac7d.png)
![Moped-Dashboard](https://github-production-user-asset-6210df.s3.amazonaws.com/1463708/269081011-e2428318-fe74-4110-9de1-e69783aa37c1.png)

<img width="1197" alt="Screen Shot 2022-10-25 at 9 48 20 AM" src="https://user-images.githubusercontent.com/14793120/197800486-f101ba2f-a672-493d-acc7-fe61be4f8aa1.png">

---

This Github issue represents a project of Austin Transportation's [Data & Technology Services](https://austinmobility.io/) team. Project status is documented regularly in the comments below.

**Comment by johnclary** [comment]
We have continued to deliver a large number of incremental improvements to the Moped user interface, while also staying focused on the game-changing features that will expand our Moped user base. 

Some notable incremental improvements:
- Added additional fields for tracking contract data
- Implemented better tracking of who created a project, the workgroup responsible for deliver, and a project's sponsors.
- Made the project editing experience more intuitive and reduced the number of clicks required for basic tasks
- Refined the project list layout and expanded the data that can be downloaded to a spreadsheet

**Comment by johnclary** [comment]
We have continued to deliver a large number of incremental improvements to the Moped user interface, while also staying focused on the game-changing features that will expand our Moped user base. 

### Incremental improvements

- Tracking project status alongside project status updates and comment, so that users can easily track a project's progress over time
- 

### Activity feed

- We have completed the foundational technical work to simplify how we scale out our activity feed, and are continuing to improve on the visual design of the feed, as well as add change tracking to all project properties. We most recently added tracking of changes to a project's tags.

### Project mapping

Our mapping interface overhaul is on track to be released with `v1.15` on January 19th!

### Access database migration

Our work on the legacy Access database migration is progressing smoothly. We're meeting with stakeholders on a weekly basis, while simultaneously implementing the migration code so that stakeholders can review our progress and give feedback.

**Comment by johnclary** [comment]
### Recently completed
- New project mapping interface ✨ 
- Better activity log tracking

### In progress (ETA 3/1)
- Show child projects on project map
- Show signal ID on project map list
- Project component phasing

### On deck / ongoing
- Access Database migration
- Enhanced project work type tracking
- New component and subcomponent options

**Comment by johnclary** [comment]
We have implemented a number of new features which support cross-division collaboration, with a focus on spatial data quality. We've also implemented a major upgrade of our state-of-the-art user interface library, [Material UI](https://mui.com/).

### Recently completed
- Show child projects on project map
- Show signal IDs on project map list
- Enable project component phases

### In progress (ETA 3/1)
- Active Transportation & Street Design Access Database migration
- Enhanced project work type tracking
- New component and subcomponent options

### Scoping
- All projects map


Screenshot of related projects shown on map:
<img width="1505" alt="Screenshot 2023-05-01 at 10 10 20 AM" src="https://user-images.githubusercontent.com/14793120/235465099-9b5041f8-f6b4-4f21-9d07-a491d9baad5b.png">

Screenshot of all projects map prototype:
<img width="1512" alt="Screenshot 2023-05-01 at 10 09 38 AM" src="https://user-images.githubusercontent.com/14793120/235465215-831a4a08-a315-4536-868f-96f723bbd499.png">

**Comment by johnclary** [comment]
We have continued to push minor bug fixes while keeping our focus on the Active Transportation database migration. We are running successful end-to-end tests of the migration, with just a few features in the Access database left to address.

### Recently completed
- Upgrade User Interface library to latest version
- Add support for tagging project components
- Add support for tracking project component work types
- Create Moped components from ArcGIS Online features

### In progress
- Add support for enhanced work authorization and contract tracking

### Scoping
- All projects map

**Comment by johnclary** [comment]
We have continued to push minor bug fixes while keeping our focus on the Active Transportation & Street Design (ATSD) database migration. We are running successful end-to-end tests of the migration, with just a few features in the Access database left to address.

### Recently completed
- Add support for enhanced work authorization and contract tracking (to support ATSD migration)

### In progress
- PowerBI and ArcGIS Online Integration

### Scoping
- All projects map

**Comment by johnclary** [comment]
The development work to migrate Active Transportation & Street Design (ATSD) from their Access DB to Moped is 95% complete. We will begin quality assurance testing next week. It is possible that we will complete the migration and go live with ATSD in Moped in September! This will include 12k+ project records plus 30 new users in the Moped system.

Additionally, we are supporting the Smart Mobility Office's effort to onboard to Moped. They are currently testing Moped for managing their pilot projects

### Recently completed

- Enhanced map editing features for complex projects
- Expanded options list for subcomponents, work types, and tags
- Expanded options for project funding and programs
- Create 70+ Moped users for ATSD-related projects (includes active and historical users)

### In progress
- PowerBI and ArcGIS Online Integration
- Overhauled user interface for tracking work activities
- Data modeling for ATSD-specific project tags and project milestones
- Ability to store links to files on G drive, sharepoint, etc

### Scoping
- All projects map

**Comment by johnclary** [comment]
Quality assurance testing is underway as we work to migrate the Active Transportation & Street Design (ATSD) from their Access DB to Moped. As a result of this recent QA testing, our team has quickly delivered a few must-have features for the ATSD. Our expectation is that the migration will be complete this month (Oct 2023).


### Recently completed

- The ability to copy a project component to another project
- Improvements to project location tracking
- Ability to store links to files on G drive, sharepoint, etc
- Project list performance improvements
- Overhauled user interface for tracking work activities
- Data modeling for ATSD-specific project tags and project milestones

### In progress
- PowerBI and ArcGIS Online Integration
- Ability to hyperlink to a project component
- Add additional fields to the advanced project search, plus more performance enhancements

### Scoping
- All projects map

**Comment by johnclary** [comment]
We successfully completed the ATSD Access DB migration on Nov 6th. With all 2,000+ projects and 11,000+ project components successfully uploaded to Moped, the Active Transportation & Street Design Division is now live in Moped!

We are continuing to prioritize various enhancements which provide reporting continuity for ATSD. Those issues are tracked under the epic #13976.

It is also worth noting that we completed the Moped <> ArcGIS Online integration (#10203) and released many substantial improvements in the [v1.35 Release](https://github.com/cityofaustin/atd-moped/releases/tag/v1.35.0) that preceded the Access DB migration.
